pkg/clients/release/plans.go

```diff
@@ -15,7 +15,7 @@ import (
 )
 
 // CreateReleasePlan creates a new ReleasePlan using the given parameters.
-func (r *ReleaseController) CreateReleasePlan(name, namespace, application, targetNamespace, autoReleaseLabel string, data *runtime.RawExtension) (*releaseApi.ReleasePlan, error) {
+func (r *ReleaseController) CreateReleasePlan(name, namespace, application, targetNamespace, autoReleaseLabel string, data *runtime.RawExtension, tenantPipeline *tektonutils.ParameterizedPipeline) (*releaseApi.ReleasePlan, error) {
```

Comment:
This function is a function in common lib that is used in the whole repo.  It need to be changed since the definition of ReleasePlan CR changed.  Right?

Reply:
I understand. The problem is that for _TenantPipelines_ the Pipeline field of the releasePlan *should* be `nil`. The way you wrote, it won't be and would cause this error https://github.com/konflux-ci/release-service/blob/c6133cc0b908743ebd1f3a92c7ff9b123353b97b/controllers/release/adapter.go#L712

It is fine having the parameter there, but the CR should have  be `Pipeline != nil`  for normal Pipelines.

Reply:
In most the test case, the CR doesn't need to be checked.

Reply:
My previous comment iis the opposite. For tenant pipelines the `Pipeline` field of the releasePlan should not be nil. I was thinking one thing and wrote the opposite. Anyhow, this LGTM.